### PR TITLE
fix(config): auto detection and default to tls

### LINKED ISSUE #40: tls config requirement
I was using version 1.0.1 to retrieve secrets successfully but then when I built a new image with 1.1.0 it started to fail with a timeout trying to reach the secretsmanager endpoint over http:

failed to connect to 'http://secretsmanager.us-west-2.amazonaws.com:80'

I had not changed any configuration at all. I went looking in your unit tests and saw the extra change: 

config.tls = true

If this is required now you might want to update your public examples to reflect this requirement. 

Thanks for the great library.

### PR DESCRIPTION
**Pull request by StarlightIbuki**
fix #40

### CODE REVIEW

**Review by windmgc — APPROVED** [review]

File: src/resty/aws/request/build.lua [hunk 1]
```diff
@@ -73,16 +73,21 @@ local function parse_query(q)
...
+      scheme = scheme or endpoint:sub(1, s-1):lower()
+      endpoint = endpoint:sub(e+1, -1)
+    end
+
+    scheme = scheme or "https"
+    config.scheme = scheme
+
+    -- the "globalSSL" one from the region_config_data file
+    if config.tls == nil then
+      config.tls = scheme == "https"
```
Comment (Tieske) [inline src/resty/aws/request/build.lua:90]:
```suggestion
      -- the "globalSSL" one from the region_config_data file
      scheme = scheme or endpoint:sub(1, s-1):lower()
      endpoint = endpoint:sub(e+1, -1)
    end

    scheme = scheme or "https"
    config.scheme = scheme

    if config.tls == nil then
      config.tls = scheme == "https"
```

The comment is now in the wrong position

**Review by Tieske — CHANGES_REQUESTED** [review]
minor issue with comment in the wrong position.

Also, can we add a changelog entry?

File: src/resty/aws/request/build.lua (hunk 1 above)
Comment (StarlightIbuki) [reply src/resty/aws/request/build.lua]: done

**Review by Tieske — APPROVED** [review]

**Comment by Tieske** [comment]
merged and released in 1.1.2

**Pull request merged**